crates/uv/src/commands/tool/run.rs

```diff
@@ -104,6 +110,35 @@ pub(crate) async fn run(
         return Err(anyhow::anyhow!("Tool command could not be parsed as UTF-8 string. Use `--from` to specify the package name."));
     };
 
+    if let Some(ref f) = from {
+        let possible_script = Path::new(f);
+        if is_python_script(possible_script) {
+            return Err(anyhow::anyhow!(
+                "It looks you have passed a script instead of a package into `--from`. \n\n{}{} Did you mean to run a tool with `{}`?",
+                    "hint".bold().cyan(),
+                    ":".bold(),
+                    format!("{} --from {} {}", invocation_source, PackageName::new(possible_script.to_string_lossy().to_string())?.cyan(), target),
```

Comment:
Minor note, rustfmt chokes on macros so you need to format those manually sometimes.